sopel/bot.py

```diff
@@ -211,14 +211,35 @@ def setup(self):
         self.scheduler.start()
 
     def setup_logging(self):
-        logger.setup_logging(self)
+        logger.setup_logging(self.config)
```

Comment:
@dgw the more I think of this... the more I want to move it *out* of the bot.

It's more of a "client" configuration, i.e. the tool used to launch the bot.

Do you mind if I try to put logging setup into `sopel.cli.utils`, and then I'd call it from every single command line tool?

Reply:
I think Sopel should always have logs. A lot of this rework makes sense to me, but not completely decoupling the Bot object from logs.

Reply:
If we decouple logging, why not decouple plugins? Why not make the launch tool set up job scheduling itself? You see how we get into the weeds just by removing one piece of the bot's startup?

Reply:
Yeah, the bot always *send* logs. But does it have to configure it itself? It's more a "load the settings" problem rather than a "load the bot itself".

And yes, I'd like very much to be able to load and pre-setup plugins without the bot, for many good reasons. :grin:

Reply:
Note that I'd keep the "add the channel logger" in the bot's setup. Because that is 100% bot-specific.